exec/cnex/lib/file_win32.c

```diff
@@ -0,0 +1,243 @@
+
+    BOOL r = CopyFile(filename, destination, FALSE);
+    if (!r) {
+        Cell *dest = cell_fromCString(destination);
+        handle_error(exec, GetLastError(), dest->string);
+        cell_clearCell(dest);
+    }
+    free(destination);
+    free(filename);
+}
+
+void file_delete(TExecutor *exec)
+{
+    char *filename = string_asCString(top(exec->stack)->string); pop(exec->stack);
+
+    BOOL r = DeleteFile(filename);
+    if (!r) {
+        if (GetLastError() != ERROR_FILE_NOT_FOUND) {
+            Cell *file = cell_fromCString(filename);
+            handle_error(exec, GetLastError(), file->string);
+            cell_clearCell(file);
+        }
+    }
+    free(filename);
+}
+
+void file_exists(TExecutor *exec)
+{
+    char *filename = string_asCString(top(exec->stack)->string); pop(exec->stack);
+
+    int r = _access(filename, 0) == 0;
+
+    push(exec->stack, cell_fromBoolean(r));
+    free(filename);
+}
+
+void file_files(TExecutor *exec)
+{
+    TString *path = string_fromString(top(exec->stack)->string); pop(exec->stack);
+
+    Cell *r = cell_createArrayCell(0);
+    path = string_appendCString(path, "\\*");
+    string_ensureNullTerminated(path);
+
+    WIN32_FIND_DATA fd;
+    HANDLE ff = FindFirstFile(path->data, &fd);
+    if (ff != INVALID_HANDLE_VALUE) {
+        do {
+            Cell l; cell_resetCell(&l); l.type = cString;
+            l.string = string_createCString(fd.cFileName);
+            cell_arrayAppendElement(r, l);
+            string_freeString(l.string);
+        } while (FindNextFile(ff, &fd));
+        FindClose(ff);
+    }
+
+    push(exec->stack, r);
+    string_freeString(path);
+}
+
+void file_getInfo(TExecutor *exec)
+{
+    TString *name = top(exec->stack)->string;
+    string_ensureNullTerminated(name);
+
+    WIN32_FIND_DATA fd;
+    HANDLE ff = FindFirstFile(name->data, &fd);
+    if (ff == INVALID_HANDLE_VALUE) {
+        handle_error(exec , GetLastError(), name);
+        pop(exec->stack);
+        return;
+    }
+    FindClose(ff);
+
+    // TODO: Implement OS level checks to validate if file is actually executable.  (.COM, .MSI, .HTA, .CMD, .BAT, etc.)
+    TString cexe = { 4, ".exe" };
+    Cell *r = cell_createArrayCell(0);
+    Cell t; t.type = cString;
+    // name: String
+    t.string = string_createCString(fd.cFileName);
+    cell_arrayAppendElement(r, t);
+    string_freeString(t.string);
+    // size: Number
+    t.type = cNumber; t.number = number_from_uint64(((uint64_t)(fd.nFileSizeHigh) << 32) | fd.nFileSizeLow);
+    cell_arrayAppendElement(r, t);
+    // readable: Boolean
+    t.type = cBoolean; t.boolean = TRUE;
+    cell_arrayAppendElement(r, t);
+    // writable: Boolean
+    t.type = cBoolean; t.boolean = ((fd.dwFileAttributes & FILE_ATTRIBUTE_READONLY) == 0);
+    cell_arrayAppendElement(r, t);
+    // executable: Boolean
+    t.type = cBoolean; t.boolean = name->length >= 4 && string_endsWithNoCase(name, &cexe);
```

Comment:
I'm probably going to change this to always return FALSE for the executable bit on Windows. The definition of "executable" on win32 is too squishy to be useful, and furthermore it's not a property of the file *itself* but a property of the environment in which it lives.

Reply:
Yea, I don't disagree with that.  This would be an easy change here.